### PR TITLE
[MISC] Fixing bug arising when repositories are shallow

### PR DESCRIPTION
**Pull request by deusebio**
The actions are most-likely failing across repositories, since we have introduced a check to make sure that tags do indeed belong to the base branch. 

However, git repositories in Github Actions are shallow, only carrying a single commit. Therefore, the checks may erroneously fail since the documentation tag commit is unknown to the base branch. 

While doing this I have also slightly refactored the `pre_flight_check` to provide some more meaningful logging

### CODE REVIEW

File: src/repository.py [hunk 1]
```diff
@@ -280,6 +280,7 @@ def is_commit_in_branch(self, commit_sha: str, branch: str | None = None) -> boo
         """
         star_pattern = re.compile(r"^\* ")
         try:
+            self._git_repo.git.fetch("--depth=2147483647")
```
Comment (deusebio) [inline src/repository.py:283]: > The special depth 2147483647 (or 0x7fffffff, the largest positive number a signed 32-bit integer can contain) means infinite depth.

Reference [here](https://git-scm.com/docs/shallow)

Comment (jdkandersson) [reply src/repository.py:287]: Perhaps make this a comment on the code?

**Review by jdkandersson — APPROVED** [review]

File: src/__init__.py [hunk 2]
```diff
@@ -225,10 +225,26 @@ def pre_flight_checks(clients: Clients, user_inputs: UserInputs) -> bool:
...
-        return True
+        return flag
+
+    logging.info(
+        "documentation tag %s does not exists. Creating at commit %s",
+        DOCUMENTATION_TAG,
+        clients.repository.current_commit,
+    )
+    clients.repository.tag_commit(DOCUMENTATION_TAG, clients.repository.current_commit)
+    return True
```
Comment (gregory-schiano) [inline src/__init__.py:250]: I'm wondering if the logic shouldn't be a bit refactored, something like:
```python
if not documentation_commit:
  logging.info(...)
  clients.repository.tag_commit(...)
  return True
if flag
  return True
logging.error(...)
return False
```

It reduces indentation, manages return True first, then finally the error case.
Ideally the return False should be first, but that'll probably less readable

Comment (gregory-schiano) [reply src/__init__.py]: Great, thx :)

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Review by jdkandersson — APPROVED** [review]

**Pull request merged**